File: petties-web/src/types/report.ts
export type ReportStatus = 'PENDING' | 'APPROVED' | 'REJECTED' | 'WITHDRAWN';

/** Legacy JSON shape; create/update reports use multipart in reportService. */
export interface ReportRequest {
  bookingId: string;
  reason: string;
  attachmentUrls?: string[];
}

export interface UpdateReportRequest {
  reason: string;
  attachmentUrls?: string[];
}

export interface ResolveReportRequest {
  status: ReportStatus;
  adminNote?: string;
}

export interface ReportResponse {
  id: string;
  
  // Booking Info
  bookingId: string;
  bookingCode: string;
  
  // Reporter Info
  reporterId: string;
  reporterName: string;
  reporterRole: string;
  reporterPhone?: string;
  
  // Reported User Info (if Pet Owner is reported)
  reportedUserId?: string;
  reportedUserName?: string;
  reportedUserRole?: string;
  reportedUserPhone?: string;

  // Reported Clinic Info (if Clinic is reported)
  reportedClinicId?: string;
  reportedClinicName?: string;
  reportedClinicPhone?: string;

  // Report details
  reason: string;
  attachmentUrls?: string[];
  status: ReportStatus;
  adminNote?: string;
  createdAt: string;
  updatedAt: string;
}

File: petties_mobile/lib/data/models/report.dart
class ReportResponse {
  final String id;
  final String bookingId;
  final String bookingCode;
  final String reportedName; // Map từ reportedClinicName hoặc reportedUserName
  final String reason;
  final List<String> imageUrls;
  final String status;
  final String? adminNote;
  final DateTime? createdAt;
  final DateTime? updatedAt;

  ReportResponse({
    required this.id,
    required this.bookingId,
    required this.bookingCode,
    required this.reportedName,
    required this.reason,
    this.imageUrls = const [],
    required this.status,
    this.adminNote,
    this.createdAt,
    this.updatedAt,
  });

  factory ReportResponse.fromJson(Map<String, dynamic> json) {
    List<String> parsedImages = [];
    if (json['attachmentUrls'] != null) {
      parsedImages = List<String>.from(json['attachmentUrls']);
    }

    String reported = json['reportedClinicName'] ?? json['reportedUserName'] ?? 'Đối tượng không xác định';

    return ReportResponse(
      id: json['id']?.toString() ?? '',
      bookingId: json['bookingId']?.toString() ?? '',
      bookingCode: json['bookingCode']?.toString() ?? '',
      reportedName: reported,
      reason: json['reason'] ?? '',
      status: json['status'] ?? 'PENDING',
      adminNote: json['adminNote']?.toString(),
      imageUrls: parsedImages,
      createdAt: json['createdAt'] != null ? DateTime.tryParse(json['createdAt']) : null,
      updatedAt: json['updatedAt'] != null ? DateTime.tryParse(json['updatedAt']) : null,
    );
  }

  String get statusText {
    switch (status) {
      case 'PENDING':
        return 'Chờ xử lý';
      case 'PROCESSING':
        return 'Đang xử lý';
      case 'RESOLVED':
        return 'Đã xử lý';
      case 'CANCELLED':
        return 'Đã hủy';
      case 'REJECTED':
        return 'Từ chối giải quyết';
      default:
        return status;
    }
  }
}
